src/StorageWriter/ClaimController.ts

```diff
@@ -1,54 +1,83 @@
 import { Claim } from '@po.et/poet-js'
 import { inject, injectable } from 'inversify'
-import { Collection, Db } from 'mongodb'
 import * as Pino from 'pino'
+import { pipeP } from 'ramda'
 
 import { childWithFileName } from 'Helpers/Logging'
-import { Messaging } from 'Messaging/Messaging'
 
-import { ExchangeConfiguration } from './ExchangeConfiguration'
+import { DAOClaims } from './DAOClaims'
 import { IPFS } from './IPFS'
 
+enum LogTypes {
+  info = 'info',
+  trace = 'trace',
+  error = 'error',
+}
+
+interface StoreNextClaimData {
+  readonly claim: Claim
+  readonly ipfsFileHash?: string
+}
+
 @injectable()
 export class ClaimController {
   private readonly logger: Pino.Logger
-  private readonly db: Db
-  private readonly collection: Collection
-  private readonly messaging: Messaging
+  private readonly daoClaims: DAOClaims
   private readonly ipfs: IPFS
-  private readonly exchange: ExchangeConfiguration
 
   constructor(
     @inject('Logger') logger: Pino.Logger,
-    @inject('DB') db: Db,
-    @inject('Messaging') messaging: Messaging,
-    @inject('IPFS') ipfs: IPFS,
-    @inject('ExchangeConfiguration') exchange: ExchangeConfiguration
+    @inject('DAOClaims') daoClaims: DAOClaims,
+    @inject('IPFS') ipfs: IPFS
   ) {
     this.logger = childWithFileName(logger, __filename)
-    this.db = db
-    this.collection = this.db.collection('storageWriter')
-    this.messaging = messaging
+    this.daoClaims = daoClaims
     this.ipfs = ipfs
-    this.exchange = exchange
   }
 
-  async create(claim: Claim): Promise<void> {
+  private readonly log = (level: LogTypes) => (message: string) => async (value: any) => {
+    const logger = this.logger
+    logger[level]({ value }, message)
+    return value
+  }
+
+  public readonly create = async (claim: Claim): Promise<void> => {
     const logger = this.logger.child({ method: 'create' })
 
-    logger.trace({ claim }, 'Storing Claim')
+    logger.trace({ claim }, 'Adding Claim')
 
-    const ipfsFileHash = await this.ipfs.addText(JSON.stringify(claim))
+    await this.daoClaims.addClaim(claim)
```

Comment:
would be useful to add a `logger.trace` when finish to add the claim?

Reply:
added